File: android/src/main/kotlin/flutter/curiosity/CuriosityEvent.kt
package flutter.curiosity

import android.os.Handler
import android.os.Looper
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

class CuriosityEvent(binaryMessenger: BinaryMessenger) : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null
    private var eventChannel: EventChannel? = null
    private val handler = Handler(Looper.getMainLooper())

    init {
        eventChannel = EventChannel(binaryMessenger, "curiosity/event")
        eventChannel!!.setStreamHandler(this)
    }


    fun sendEvent(arguments: Any?) {
        handler.post {
            eventSink?.success(arguments)
        }
    }

    fun dispose() {
        eventSink?.endOfStream()
        eventSink = null
        eventChannel?.setStreamHandler(null)
        eventChannel = null
    }

    override fun onListen(arguments: Any?, event: EventChannel.EventSink?) {
        eventSink = event
    }

    override fun onCancel(arguments: Any?) {
        dispose();
    }
}

File: ios/Classes/CuriosityEvent.swift
import Flutter
import Foundation

class CuriosityEvent: NSObject, FlutterStreamHandler {
    var eventSink: FlutterEventSink?
    var eventChannel: FlutterEventChannel?
    
    init(messenger: FlutterBinaryMessenger) {
        super.init()
        print("已经初始化了CuriosityEvent")
        eventChannel = FlutterEventChannel(name: "curiosity/event", binaryMessenger: messenger)
        eventChannel?.setStreamHandler(self)
    }
    
    internal func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        eventSink = events
        return nil
    }
    
    func sendEvent(arguments: Any?) {
        eventSink?(arguments)
    }
    
    func dispose() {
        eventSink = nil
        eventChannel?.setStreamHandler(nil)
        eventChannel = nil
    }
    
    internal func onCancel(withArguments arguments: Any?) -> FlutterError? {
        dispose()
        return nil
    }
}
